core/src/main/java/com/yahoo/ycsb/BasicDB.java

```diff
@@ -111,21 +124,23 @@ public Status read(String table, String key, Set<String> fields, HashMap<String,
 
 		if (verbose)
 		{
-			System.out.print("READ "+table+" "+key+" [ ");
+      StringBuilder sb = getStringBuilder();
+			sb.append("READ ").append(table).append(" ").append(key).append(" [ ");
 			if (fields!=null)
 			{
 				for (String f : fields)
 				{
-					System.out.print(f+" ");
+          sb.append(f).append(" ");
 				}
 			}
 			else
 			{
-				System.out.print("<all fields>");
+        sb.append("<all fields>");
 			}
 
-			System.out.println("]");
-		}
+      sb.append("]");
+      System.out.println(sb);
```

Comment:
Does anything say that individual calls to S.o.println are threadsafe? Or are we just decreasing the odds of a collision?